specs/bond-manager.md

```diff
@@ -0,0 +1,100 @@
+# Bond Manager Interface
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Overview](#overview)
+- [The Bond Problem](#the-bond-problem)
+  - [Simple Bond](#simple-bond)
+  - [Variable Bond](#variable-bond)
+- [Types of Bond Managers](#types-of-bond-managers)
+  - [OracleBondManager](#oraclebondmanager)
+  - [AttestationBondManager](#attestationbondmanager)
+  - [FaultBondManager](#faultbondmanager)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The bond manager contracts handle bonds for both the [dispute-games](./dispute-game.md)
+and the `L2OutputOracle` contract (detailed in [propoals](./proposals.md)).
+
+When outputs are permissionlessly posted to the `L2OutputOracle`, the proposer must
+submit some ether as a "bond" to disincentivize spam and invalid outputs. These bonds
+are handled by the `L2OutputOracle`'s bond manager. The `L2OutputOracle` contract should
+forward these bonds on proposals, and call bonds when outputs are either deleted or
+finalized.
+
+Similarly, the various types of [dispute-games](./dispute-game-interface.md) require
+different bond management. In the simplest dispute game (the "Attestation" dispute game)
+bonds will not be required since the attestors are permissioned. But in more complex games, such as the
+fault dispute game, bonds will be required since the challengers and defenders perform a series of
+alternating onchain transactions requiring bonds at each step. In this case, a separate bond manager than
+that of the `L2OutputOracle` is required.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and returns
+it when finalized or seized. But the uncertainty of introducing bonds lies in the
+bond _sizing_, i.e. how much should a bond be? Sizing bonds correctly entails a variety of
+tradeoffs. Price them too high, and not enough proposers will post outputs. Price them too low, and
+challengers won't be incentivized to delete outputs if the gas cost of doing so outways the bond itself.
+
+Below, we outline two different approaches to sizing bonds and the tradeoffs of each.
+
+### Simple Bond
```

Comment:
I think we should try to describe the type of bond independent of the thing that it is contesting (the L2 Output Oracle). We want to say that there are some invariants that must be held for this to be incentive compatible:
- known cost of progressing the game
- known upper bound on cost of completing the game

The size of the bond can be computed up front when the dispute game is created. We define a linear function that is something like: `number of signers * amount of gas used to progress game (add signature) + security overhead`. We could also multiply in the security overhead but made it simple to not deal with decimals